decidim-comments/spec/commands/create_comment_spec.rb

```diff
@@ -70,49 +70,16 @@ module Comments
             end.to change(Comment, :count).by(1)
           end
 
-          it "sends a notification to the corresponding users except the comment's author" do
-            follower = create(:user, organization: organization)
-
-            expect(commentable)
-              .to receive(:users_to_notify_on_comment_created)
-              .and_return([follower, author])
-
-            expect_any_instance_of(Decidim::Comments::Comment) # rubocop:disable RSpec/AnyInstance
-              .to receive(:id).at_least(:once).and_return 1
-
-            expect_any_instance_of(Decidim::Comments::Comment) # rubocop:disable RSpec/AnyInstance
-              .to receive(:root_commentable).at_least(:once).and_return commentable
-
-            expect(Decidim::EventsManager)
-              .to receive(:publish)
-              .with(
-                event: "decidim.events.comments.comment_created",
-                event_class: Decidim::Comments::CommentCreatedEvent,
-                resource: commentable,
-                recipient_ids: [follower.id],
-                extra: {
-                  comment_id: a_kind_of(Integer)
-                }
-              )
+          it "sends the notifications" do
+            creator_double = double(create: true)
 
-            command.call
-          end
+            expect(NewCommentNotificationCreator)
+              .to receive(:new)
+              .with(kind_of(Comment), [])
```

Comment:
Maybe instead of this intensive mocking we can assert that `Decidim::EventsManager.publish` is being called?

Reply:
Mhh not sure, I'd rather make sure the whole chain is being tested. `Decidim::EventsManager.publish` is being tested from within the `NewCommentNotificationCreator` tests